backend/courses/models.py

```diff
@@ -1419,3 +1419,40 @@ def create_or_update_user_profile(sender, instance, created, **kwargs):
     if created and instance.email != "":
         instance.profile.email = instance.email
     instance.profile.save()
+
+
+
+class Friendship(models.Model):
+    """
+    Used to track friendships along with requests status 
+    """
+    
+    requestSender = models.ForeignKey(
+        UserProfile,
+        on_delete=models.CASCADE,
+        related_name="sender",
+        help_text="The person (user) who sent the request.",
+    )
+
+    requestReciever = models.ForeignKey(
```

Comment:
This is _super_ pedantic, but we might want to call this `request_recipient` (as a receiver is usually a device that handles receiving, as opposed to a person receiving a single request).